### ISSUE TITLE
Crash in FormDataIODevice::reset()

### ISSUE THREAD

**Issue by annulen** [annulen/webkit#347]
1. In https://codereview.qt-project.org/#/c/103087/2/Source/WebCore/platform/network/qt/QNetworkReplyHandler.cpp null check was lost for `formData`
2. `formData` can be null, as it is taken from `ResourceRequest::httpBody()`
3. Check all upload and blob-related failing/crashing/skipped tests, check `http/tests/local/blob/send-sliced-data-blob.html` and `http/tests/local/blob/send-data-blob.html` (see fac575c24a9e2a6ee11d18533c8b4960cb550397)
4. `m_formData` should be converted to reference (not for backport!)

Backtrace reported by @haasn:
```
Program terminated with signal SIGSEGV, Segmentation fault.
#0  0x00007fd524a573c9 in raise () from /lib64/libpthread.so.0
[Current thread is 1 (Thread 0x7fd525341700 (LWP 885))]
(gdb) bt
#0  0x00007fd524a573c9 in raise () from /lib64/libpthread.so.0
#1  <signal handler called>
#2  0x00007fd5164997a9 in WTF::Vector<WebCore::FormDataElement, 0ul, WTF::CrashOnOverflow, 16ul>::operator=(WTF::Vector<WebCore::FormDataElement, 0ul, WTF::CrashOnOverflow, 16ul> const&) () from /usr/lib64/libQt5WebKit.so.5
#3  0x00007fd51649a17f in WebCore::FormDataIODevice::reset() () from /usr/lib64/libQt5WebKit.so.5
#4  0x00007fd51fdbfdb2 in QNonContiguousByteDeviceIoDeviceImpl::reset() () from /usr/lib64/libQt5Core.so.5
#5  0x00007fd514f0eed5 in QNetworkReplyHttpImplPrivate::resetUploadDataSlot(bool*) () from /usr/lib64/libQt5Network.so.5
#6  0x00007fd514fc14f0 in QNetworkReplyHttpImpl::qt_static_metacall(QObject*, QMetaObject::Call, int, void**) () from /usr/lib64/libQt5Network.so.5
#7  0x00007fd51feb3a81 in QObject::event(QEvent*) () from /usr/lib64/libQt5Core.so.5
#8  0x00007fd51adf8aec in QApplicationPrivate::notify_helper(QObject*, QEvent*) () from /usr/lib64/libQt5Widgets.so.5
#9  0x00007fd51ae00df6 in QApplication::notify(QObject*, QEvent*) () from /usr/lib64/libQt5Widgets.so.5
#10 0x00007fd51b551e4e in sipQApplication::notify(QObject*, QEvent*) () from /usr/lib64/python3.5/site-packages/PyQt5/QtWidgets.so
#11 0x00007fd51fe8510a in QCoreApplication::notifyInternal2(QObject*, QEvent*) () from /usr/lib64/libQt5Core.so.5
#12 0x00007fd51fe8803a in QCoreApplicationPrivate::sendPostedEvents(QObject*, int, QThreadData*) () from /usr/lib64/libQt5Core.so.5
#13 0x00007fd51fedeea6 in postEventSourceDispatch(_GSource*, int (*)(void*), void*) () from /usr/lib64/libQt5Core.so.5
#14 0x00007fd51ea35537 in g_main_context_dispatch () from /usr/lib64/libglib-2.0.so.0
#15 0x00007fd51ea35798 in g_main_context_iterate.isra () from /usr/lib64/libglib-2.0.so.0
#16 0x00007fd51ea3589b in g_main_context_iteration () from /usr/lib64/libglib-2.0.so.0
#17 0x00007fd51fedef4e in QEventDispatcherGlib::processEvents(QFlags<QEventLoop::ProcessEventsFlag>) () from /usr/lib64/libQt5Core.so.5
#18 0x00007fd51fe83a0a in QEventLoop::exec(QFlags<QEventLoop::ProcessEventsFlag>) () from /usr/lib64/libQt5Core.so.5
#19 0x00007fd51fe8c61b in QCoreApplication::exec() () from /usr/lib64/libQt5Core.so.5
#20 0x00007fd51b5504ba in meth_QApplication_exec_ () from /usr/lib64/python3.5/site-packages/PyQt5/QtWidgets.so
#21 0x00007fd524d20d21 in PyCFunction_Call () from /usr/lib64/libpython3.5m.so.1.0
#22 0x00007fd524da9074 in PyEval_EvalFrameEx () from /usr/lib64/libpython3.5m.so.1.0
#23 0x00007fd524da8947 in PyEval_EvalFrameEx () from /usr/lib64/libpython3.5m.so.1.0
#24 0x00007fd524da8947 in PyEval_EvalFrameEx () from /usr/lib64/libpython3.5m.so.1.0
#25 0x00007fd524da8947 in PyEval_EvalFrameEx () from /usr/lib64/libpython3.5m.so.1.0
#26 0x00007fd524daad14 in _PyEval_EvalCodeWithName () from /usr/lib64/libpython3.5m.so.1.0
#27 0x00007fd524daae32 in PyEval_EvalCodeEx () from /usr/lib64/libpython3.5m.so.1.0
#28 0x00007fd524daae8a in PyEval_EvalCode () from /usr/lib64/libpython3.5m.so.1.0
#29 0x00007fd524dce423 in run_mod () from /usr/lib64/libpython3.5m.so.1.0
#30 0x00007fd524dd0cc6 in PyRun_FileExFlags () from /usr/lib64/libpython3.5m.so.1.0
#31 0x00007fd524dd0e65 in PyRun_SimpleFileExFlags () from /usr/lib64/libpython3.5m.so.1.0
#32 0x00007fd524deacd4 in Py_Main () from /usr/lib64/libpython3.5m.so.1.0
#33 0x000055dd14154bff in main ()
(gdb) 
```

**Comment by annulen** [comment]
```
(gdb) bt
#0  0x00007f314e3f93c9 in raise () from /lib64/libpthread.so.0
#1  <signal handler called>
#2  0x00007f313fd94409 in WTF::Vector<WebCore::FormDataElement, 0ul, WTF::CrashOnOverflow, 16ul>::operator= (this=this@entry=0x560cc9165fa0, other=...) at /usr/src/debug/dev-qt/qtwebkit-6.0/qtwebkit-tp4/Source/WTF/wtf/Vector.h:837
#3  0x00007f313fd94ddf in WebCore::FormDataIODevice::reset (this=0x560cc9165f90) at /usr/src/debug/dev-qt/qtwebkit-6.0/qtwebkit-tp4/Source/WebCore/platform/network/qt/QNetworkReplyHandler.cpp:72
#4  0x00007f3149761db2 in QNonContiguousByteDeviceIoDeviceImpl::reset() () from /usr/lib64/libQt5Core.so.5
#5  0x00007f313e2a3ed5 in QNetworkReplyHttpImplPrivate::resetUploadDataSlot(bool*) () from /usr/lib64/libQt5Network.so.5
#6  0x00007f313e3564f0 in QNetworkReplyHttpImpl::qt_static_metacall(QObject*, QMetaObject::Call, int, void**) () from /usr/lib64/libQt5Network.so.5
#7  0x00007f3149855a81 in QObject::event(QEvent*) () from /usr/lib64/libQt5Core.so.5
#8  0x00007f314479aaec in QApplicationPrivate::notify_helper(QObject*, QEvent*) () from /usr/lib64/libQt5Widgets.so.5
#9  0x00007f31447a2df6 in QApplication::notify(QObject*, QEvent*) () from /usr/lib64/libQt5Widgets.so.5
#10 0x00007f3144ef3e4e in sipQApplication::notify(QObject*, QEvent*) () from /usr/lib64/python3.5/site-packages/PyQt5/QtWidgets.so
#11 0x00007f314982710a in QCoreApplication::notifyInternal2(QObject*, QEvent*) () from /usr/lib64/libQt5Core.so.5
#12 0x00007f314982a03a in QCoreApplicationPrivate::sendPostedEvents(QObject*, int, QThreadData*) () from /usr/lib64/libQt5Core.so.5
#13 0x00007f3149880ea6 in postEventSourceDispatch(_GSource*, int (*)(void*), void*) () from /usr/lib64/libQt5Core.so.5
#14 0x00007f31483d7537 in g_main_context_dispatch () from /usr/lib64/libglib-2.0.so.0
#15 0x00007f31483d7798 in g_main_context_iterate.isra () from /usr/lib64/libglib-2.0.so.0
#16 0x00007f31483d789b in g_main_context_iteration () from /usr/lib64/libglib-2.0.so.0
#17 0x00007f3149880f4e in QEventDispatcherGlib::processEvents(QFlags<QEventLoop::ProcessEventsFlag>) () from /usr/lib64/libQt5Core.so.5
#18 0x00007f3149825a0a in QEventLoop::exec(QFlags<QEventLoop::ProcessEventsFlag>) () from /usr/lib64/libQt5Core.so.5
#19 0x00007f314982e61b in QCoreApplication::exec() () from /usr/lib64/libQt5Core.so.5
#20 0x00007f3144ef24ba in meth_QApplication_exec_ () from /usr/lib64/python3.5/site-packages/PyQt5/QtWidgets.so
#21 0x00007f314e6c2d21 in PyCFunction_Call () from /usr/lib64/libpython3.5m.so.1.0
#22 0x00007f314e74b074 in PyEval_EvalFrameEx () from /usr/lib64/libpython3.5m.so.1.0
#23 0x00007f314e74a947 in PyEval_EvalFrameEx () from /usr/lib64/libpython3.5m.so.1.0
#24 0x00007f314e74a947 in PyEval_EvalFrameEx () from /usr/lib64/libpython3.5m.so.1.0
#25 0x00007f314e74a947 in PyEval_EvalFrameEx () from /usr/lib64/libpython3.5m.so.1.0
#26 0x00007f314e74cd14 in _PyEval_EvalCodeWithName () from /usr/lib64/libpython3.5m.so.1.0
#27 0x00007f314e74ce32 in PyEval_EvalCodeEx () from /usr/lib64/libpython3.5m.so.1.0
#28 0x00007f314e74ce8a in PyEval_EvalCode () from /usr/lib64/libpython3.5m.so.1.0
#29 0x00007f314e770423 in run_mod () from /usr/lib64/libpython3.5m.so.1.0
#30 0x00007f314e772cc6 in PyRun_FileExFlags () from /usr/lib64/libpython3.5m.so.1.0
#31 0x00007f314e772e65 in PyRun_SimpleFileExFlags () from /usr/lib64/libpython3.5m.so.1.0
#32 0x00007f314e78ccd4 in Py_Main () from /usr/lib64/libpython3.5m.so.1.0
#33 0x0000560cbbe47bff in main ()
```

**Comment by annulen** [comment]
Fixed in e2fabcf1684a591653e31fbc44164cb7db723284
